c_src/eleveldb.cc

```diff
@@ -568,8 +613,13 @@ async_open(
     eleveldb_priv_data& priv = *static_cast<eleveldb_priv_data *>(enif_priv_data(env));
 
     leveldb::Options *opts = new leveldb::Options;
+    opts->expiry_module.assign(priv.m_ExpiryModule.get());
+
     fold(env, argv[2], parse_open_option, *opts);
+
     opts->fadvise_willneed = priv.m_Opts.m_FadviseWillNeed;
+    if (NULL==priv.m_ExpiryModule.get() && NULL!=opts->expiry_module.get())
```

Comment:
Line 616:  has a previous vnode opened and created an expiry module?  Give me the previously created object if it exists.
Line 621:  if there was no previously created expiry module and one suddenly appears due to this fold, make it the default.

This squirrel code defends against the future possibility of having different expiry modules by vnode and yet still a global default.  Not likely ... but possible given that the future code is not yet written.